blob/service.go

```diff
@@ -51,14 +51,32 @@ func NewService(
 	}
 }
 
+// SubmitOptions contains the information about fee and gasLimit price in order to configure the Submit request.
+type SubmitOptions struct {
+	Fee      int64
+	GasLimit uint64
+}
+
+// DefaultSubmitOptions creates a default fee and gas price values.
+func DefaultSubmitOptions() *SubmitOptions {
+	return &SubmitOptions{
+		Fee:      -1,
+		GasLimit: 0,
+	}
+}
+
 // Submit sends PFB transaction and reports the height in which it was included.
 // Allows sending multiple Blobs atomically synchronously.
 // Uses default wallet registered on the Node.
 // Handles gas estimation and fee calculation.
-func (s *Service) Submit(ctx context.Context, blobs []*Blob) (uint64, error) {
+func (s *Service) Submit(ctx context.Context, blobs []*Blob, options *SubmitOptions) (uint64, error) {
```

Comment:
I'm not sure, we have enough time/capacity to fix variadics now. We can go with `SubmitOptions` and once the variadics are fixed, we can simply remove `nil`. Or users can use `DefaultOptions` and then nothing should be removed at all.

I agree that removing the whole endpoint will be harder

Reply:
Yet another suggestion:
Keep only version with options, but make options a private struct. It will reduce ux to only 2 cases: defaults, manually set by constructor:
```
type submitOptions struct {
	fee      int64
	gasLimit uint64
}

func DefaultSubmitOptions() submitOptions {
	return submitOptions{
		fee:      -1,
		gasLimit: 0,
	}
}

func SubmitOptions(fee, gaslimit int) submitOptions{
	return submitOptions{
		fee:      -1,
		gasLimit: 0,
	}
}
```

Reply:
Also sounds good to me.

Reply:
Just realised that private field will not work as we are marshaling/unmarshaling them during rpc calls

Reply:
Relying on passing nil is something golang specific, and this will be ugly in the rust version where you will always need to instantiate the struct. I would much much rather have two methods, it is also less breaking. I absolutely love having a method that doesn't take any parameters

Reply:
Still don't like the approach where you have to specify a new endpoint once you've faced some edge cases or you need a possibility to configure the request. If we accept this pattern everywhere in our code, the API can significantly grow in the future imo